p2p/src/authenticated/actors/peer/actor.rs

```diff
@@ -237,17 +239,21 @@ impl<E: Spawner + Clock + ReasonablyRealtime + Rng + CryptoRng + Metrics, P: Arr
                                         ))
                                         .inc();
                                     let wait = negative.wait_time_from(context.now());
-                                    context.sleep(wait).await;
+                                    context
+                                        .sleep(wait)
+                                        .instrument(info_span!("rate_limit_sleep"))
```

Comment:
For async calls, we need to be very careful when using `span.enter()`: https://docs.rs/tracing/latest/tracing/struct.Span.html#in-asynchronous-code

Reply:
I believe using `#[instrument]` may already do this but its much tougher to use in a loop like this (as we'd have to break everything into sub-functions I think)